ydb/docs/en/core/deploy/configuration/config.md

```diff
@@ -456,6 +456,109 @@ actor_system_config:
 | `progress_threshold` | The actor system supports requesting message sending scheduled for a later point in time. The system might fail to send all scheduled messages at some point. In this case, it starts sending them in "virtual time" by handling message sending in each loop over a period that doesn't exceed the `progress_threshold` value in microseconds and shifting the virtual time by the `progress_threshold` value until it reaches real time. |
 | `resolution` | When making a schedule for sending messages, discrete time slots are used. The slot duration is set by the `resolution` parameter in microseconds. |
 
+## Memory controller {#memory-controller}
+
+There are many components inside {{ ydb-short-name }} [database nodes](../../concepts/glossary.md#database-node) that utilize memory. Most of them need a fixed amount, but some are flexible and can use varying amounts of memory, typically to improve performance. If such components allocate more memory than is physically available, the operating system is likely to [terminate](https://en.wikipedia.org/wiki/Out_of_memory#Recovery) the entire {{ ydb-short-name }} node, which is undesirable. The memory controller's goal is to allow {{ ydb-short-name }} to avoid out-of-memory situations while still efficiently using the available memory.
+
+Examples of components managed by the memory controller:
+
+- [Shared cache](../../concepts/glossary.md#shared-cache): stores recently accessed data pages read from [distributed storage](../../concepts/glossary.md#distributed-storage) to reduce disk I/O and accelerate data retrieval.
+- [MemTable](../../concepts/glossary.md#memtable): holds data that has not yet been flushed to [SST](../../concepts/glossary.md#sst).
+- [KQP](../../concepts/glossary.md#kqp): stores intermediate query results.
+- Allocator caches: keep memory blocks that have been released but not yet returned to the operating system.
+
+Memory limits can be configured to control overall memory usage, ensuring the database operates efficiently within the available resources.
+
+### Hard memory limit {#hard-memory-limit}
+
+The hard memory limit specifies the total amount of available memory.
+
+By default, the hard memory limit is set to the {{ ydb-short-name }} node process's [cgroups](https://en.wikipedia.org/wiki/Cgroups) memory limit.
+
+In environments without a cgroups memory limit, the default hard memory limit is equal to the host's total available memory. This allows the database to utilize all available resources in unrestricted environments, though it may lead to resource contention with other processes running on the same host.
```

Comment:
"the default hard memory limit is equal to the host's total available memory"

I would expect something like "total_available_memory * 0.9", because some other process may be running

Reply:
Actually, the real behaviour is a bit more complicated, because we also take into account how much memory is consumed outside of YDB process and add it too.

But I'm not sure if it's worth mentioning here

Reply:
Added

> In environments without a cgroups memory limit, the default hard memory limit equals to the host's total available memory. This configuration allows the database to utilize all available resources but may lead to resource competition with other processes on the same host. Although the memory controller **attempts to account for this external consumption**, such a setup is not recommended.